src/html/HTMLWriter.jl

```diff
@@ -713,6 +713,91 @@ function SearchRecord(ctx, navnode, node::Node, ::MarkdownAST.AbstractElement)
     return SearchRecord(ctx, navnode; text = mdflatten(node))
 end
 
+
+# For @example blocks
+function SearchRecord(ctx::HTMLContext, navnode::Documenter.NavNode, node::MarkdownAST.Node{Nothing}, element::Documenter.MultiOutput)
+    output_text = ""
+    
+    first_child = true
+    for child in node.children
+        if first_child
+            first_child = false
+            continue
+        end
+        
+        if isa(child.element, Documenter.MultiOutputElement) && isa(child.element.element, Dict)
+            for mime in [MIME"text/plain"(), MIME"text/markdown"(), MIME"text/html"()]
+                if haskey(child.element.element, mime)
+                    output_text *= string(child.element.element[mime]) * " "
+                    break
+                end
+            end
+        else 
+            output_text *= mdflatten(child) * " "
+        end
+    end
+    return SearchRecord(ctx, navnode; text = output_text)
+end
+
+# For @repl blocks 
+function SearchRecord(ctx::HTMLContext, navnode::Documenter.NavNode, node::MarkdownAST.Node{Nothing}, element::Documenter.MultiCodeBlock)
+    # Extract only output lines (even-numbered children)
+    output_text = ""
+    
+    # Process each child, taking only even-numbered ones (outputs)
+    child_index = 0
+    for child in node.children
+        child_index += 1
+        if child_index % 2 == 0  # Even indices are outputs
+            output_text *= mdflatten(child) * " "
+        end
+    end
+    return SearchRecord(ctx, navnode; text = output_text)
+end
+
+# For @eval blocks
+function SearchRecord(ctx, navnode, node::Node, element::Documenter.EvalNode)
+    # For @eval blocks, we only want to index the result, not the source code
+    if isnothing(element.result)
+        return SearchRecord(ctx, navnode; text = "")
+    else
+        return SearchRecord(ctx, navnode; text = mdflatten(element.result))
+    end
+end
+
+function SearchRecord(ctx, navnode, node::Node, element::Documenter.MultiOutputElement)
+    if element.element isa Dict && haskey(element.element, :source)
+        return SearchRecord(ctx, navnode; text = "")
+    end
+    
+    return SearchRecord(ctx, navnode, node, element.element)
+end
+
+"""
+Filter function to prevent duplicate search entries.
+"""
+function should_index_for_search(navnode, node, element)
```

Comment:
I think it should be sufficient to just check the type, no need to introspect into the objects? So something as simple as this should be enough probably:

```julia
exclude_from_search_index(element) = typeof(element) in (Documenter.MultiOutputElement, Documenter.MultiCodeBlock, Documenter.EvalNode)